### PR TITLE
Fixed error in Palindrome.java and gave semantic names to functions

### PR DESCRIPTION
**Pull request by TaranjeetSinghKalsi**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: src/main/java/com/thealgorithms/strings/Palindrome.java [hunk 1]
```diff
@@ -11,27 +11,27 @@ class Palindrome {
...
+            isPalindromeTwoPointer(s);
         }
 
         String[] notPalindromes = { "abb", "abc", "abc123" };
         for (String s : notPalindromes) {
-            assert !isPalindrome(s) &&
+            assert !isPalindromeStringBuilder(s) &&
             !isPalindromeRecursion(s) &&
-            !isPalindrome1(s);
+            !isPalindromeTwoPointer(s);
```
Comment (debasishbsws) [inline src/main/java/com/thealgorithms/strings/Palindrome.java:23]: Put This function into the corresponding test file.

File: src/main/java/com/thealgorithms/strings/Palindrome.java [hunk 2]
```diff
@@ -11,27 +11,27 @@ class Palindrome {
...
     /**
-     * Check if a string is palindrome string or not
+     * Check if a string is palindrome string or not using String Builder
      *
      * @param s a string to check
      * @return {@code true} if given string is palindrome, otherwise
      * {@code false}
      */
-    public static boolean isPalindrome(String s) {
+    public static boolean isPalindromeStringBuilder(String s) {
```
Comment (debasishbsws) [inline src/main/java/com/thealgorithms/strings/Palindrome.java:34]:
```suggestion
    public static boolean isPalindrome(String s) {
```
change the name to original one as it is the traditional method

**Review by debasishbsws — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/strings/Palindrome.java (hunk 2 above)
Comment (TaranjeetSinghKalsi) [reply src/main/java/com/thealgorithms/strings/Palindrome.java]: > change the name to original one as it is the traditional method

All Done. Added Test Cases, Removed main(), and changed the name of isPalindromeStringBuilder() to isPalindrome()

File: src/main/java/com/thealgorithms/strings/Palindrome.java (hunk 1 above)
Comment (TaranjeetSinghKalsi) [reply src/main/java/com/thealgorithms/strings/Palindrome.java]: done. Kindly review it @debasishbsws

**Review by debasishbsws — APPROVED** [review]
Looks good to me

**Pull request merged**